msmtools/util/sorted_schur.py

```diff
@@ -23,111 +23,64 @@ def _initialize_matrix(M, P):
-        if np.isclose(eigenval_in, np.conj(eigenval_out)):
-            block_split = True
-            warnings.warn("Clustering into " + str(m) + " clusters will split conjugate eigenvalues! "
-                          "Request one cluster more or less.")
-                
-    return top_eigenvals, block_split
-
-
-def sorted_krylov_schur(P, m, z='LM', tol=1e-16):
+
+def _check_schur(P, Q, R, eigenvalues, method = ""):
+    """Utility function to run a number of checks on the sorted schur decomposition
+    """
+
+    m = len(eigenvalues)
+
+    # check the dimensions
+    if Q.shape[1] != len(eigenvalues):
+        raise ValueError(f"Number of schur vectors does not match number of eigenvalues for method `{method}`")
+    if R.shape[0] != R.shape[1]:
+        raise ValueError(f"R is not rectangular for method `{method}`")
+    if P.shape[0] != Q.shape[0]:
+        raise ValueError(f"First dimension in P does not match first dimension in Q for method `{method}`")
+    if R.shape[0] != Q.shape[1]:
+        raise ValueError(f"First dimension in R does not match second dimension in Q for method `{method}`")
+
+    # check whether things are real
+    if not np.all(np.isreal(Q)):
+        raise TypeError(f"The orthonormal basis of the subspace returned by `{method}` is not real.",
+                        f"G-PCCA needs real basis vectors to work.")
+
+    dummy = np.dot(P, csr_matrix(Q) if issparse(P) else Q)
+    if issparse(dummy):
+        dummy = dummy.toarray()
+
+    dummy1 = np.dot(Q, np.diag(eigenvalues))
+    #     dummy2 = np.concatenate((dummy, dummy1), axis=1)
+    dummy3 = subspace_angles(dummy, dummy1)
+    #     test1 = ( ( matrix_rank(dummy2) - matrix_rank(dummy) ) == 0 )
+    test2 = np.allclose(dummy3, 0.0, atol=1e-8, rtol=1e-5)
+    test3 = (dummy3.shape[0] == m)
+    dummy4 = subspace_angles(dummy, Q)
+    test4 = np.allclose(dummy4, 0.0, atol=1e-6, rtol=1e-5)
+    if not test4:
+        raise ValueError(f"According to scipy.linalg.subspace_angles() `{method}` "
+                         f"return an invariant subspace of P. The subspace angles are: `{dummy4}`.")
+    elif not test2:
+        warnings.warn(f"According to scipy.linalg.subspace_angles() `{method}` didn't "
+                      f"return the invariant subspace associated with the top m eigenvalues, "
+                      f"since the subspace angles between the column spaces of P*Q and Q*L "
+                      f"aren't near zero (L is a diagonal matrix with the "
+                      f"sorted top eigenvalues on the diagonal). The subspace angles are: `{dummy3}`.")
+    elif not test3:
+        warnings.warn(f"According to scipy.linalg.subspace_angles() the dimension of the "
+                      f"column space of P*Q and/or Q*L is not equal to m (L is a diagonal "
+                      f"matrix with the sorted top eigenvalues on the diagonal), method = `{method}`")
+
+
+def sorted_krylov_schur(P, k, z='LM', tol=1e-16):
```

Comment:
If you use k instead of m, please also update the docstring... :)

Reply:
alright, will do!